client/src/app/users/user.service.spec.ts

```diff
@@ -61,27 +82,21 @@ describe('UserService', () => {
 
   describe('getUsers()', () => {
 
-    it('calls `api/users` when `getUsers()` is called with no parameters', () => {
-      // Assert that the users we get from this call to getUsers()
-      // should be our set of test users. Because we're subscribing
-      // to the result of getUsers(), this won't actually get
-      // checked until the mocked HTTP request 'returns' a response.
-      // This happens when we call req.flush(testUsers) a few lines
-      // down.
-      userService.getUsers().subscribe(
-        users => expect(users).toBe(testUsers)
-      );
+    it('calls `api/users` when `getUsers()` is called with no parameters', (done: DoneFn) => {
+      httpClientSpy.get.and.returnValue(asyncData(testUsers));
 
-      // Specify that (exactly) one request will be made to the specified URL.
-      const req = httpTestingController.expectOne(userService.userUrl);
-      // Check that the request made to that URL was a GET request.
-      expect(req.request.method).toEqual('GET');
-      // Check that the request had no query parameters.
-      expect(req.request.params.keys().length).toBe(0);
-      // Specify the content of the response to that request. This
-      // triggers the subscribe above, which leads to that check
-      // actually being performed.
-      req.flush(testUsers);
+      userServiceSpy.getUsers().subscribe({
+        next: users => {
+          expect(users)
+            .withContext('expected users')
+            .toEqual(testUsers);
+          done();
+        },
+        error: done.fail
+      });
+      expect(httpClientSpy.get.calls.count())
+        .withContext('one call')
+        .toBe(1);
```

Comment:
Oh, and then also, we probably want to check that `get` was called with the right arguments. (Eg, we actually are talking to `api/users` and not `http://example.com/frogs` 😜